core/src/main/java/org/mule/runtime/core/AbstractEventContext.java

```diff
@@ -143,6 +145,21 @@ public final void error(Throwable throwable) {
     }
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Publisher<Void> error(MessagingException messagingException, MessagingExceptionHandler handler) {
```

Comment:
This is part of a wider change, if we decide to not propagate MessagingExceptionHandler down though MP's.  If we do this it will be in a separate PR.